client/my-sites/site-settings/form-general.jsx

```diff
@@ -507,78 +608,83 @@ const FormGeneral = React.createClass( {
 
 		return (
 			<FormFieldset>
-				<legend>{ this.translate( 'Holiday Snow' ) }</legend>
+				<legend>{ translate( 'Holiday Snow' ) }</legend>
 				<ul>
 					<li>
 						<FormLabel>
-							<FormCheckbox name="holidaysnow" checkedLink={ this.linkState( 'holidaysnow' ) }/>
-							<span>{ this.translate( 'Show falling snow on my blog until January 4th.' ) }</span>
+							<FormCheckbox
+								name="holidaysnow"
+								checked={ !! fields.holidaysnow }
+								onChange={ this.handleCheckbox }
+							/>
+							<span>{ translate( 'Show falling snow on my blog until January 4th.' ) }</span>
 						</FormLabel>
 					</li>
 				</ul>
 			</FormFieldset>
 		);
-	},
+	}
 
 	Timezone() {
-		if ( this.props.site.jetpack ) {
+		const { fields, isRequestingSettings, site, translate } = this.props;
+		if ( site.jetpack ) {
 			return;
 		}
 
 		return (
 			<FormFieldset>
 				<FormLabel htmlFor="blogtimezone">
-					{ this.translate( 'Site Timezone' ) }
+					{ translate( 'Site Timezone' ) }
 				</FormLabel>
 
 				<Timezone
-					valueLink={ this.linkState( 'timezone_string' ) }
-					selectedZone={ this.linkState( 'timezone_string' ).value }
-					disabled={ this.state.fetchingSettings }
+					selectedZone={ fields.timezone_string }
+					disabled={ isRequestingSettings }
 					onSelect={ this.onTimezoneSelect }
 				/>
 
 				<FormSettingExplanation>
-					{ this.translate( 'Choose a city in your timezone.' ) }
+					{ translate( 'Choose a city in your timezone.' ) }
 				</FormSettingExplanation>
 			</FormFieldset>
 		);
-	},
+	}
 
 	renderJetpackSyncPanel() {
 		if ( ! config.isEnabled( 'jetpack/sync-panel' ) ) {
 			return null;
 		}
 
 		const { site } = this.props;
-		if ( ! site.jetpack || this.props.site.versionCompare( '4.2-alpha', '<' ) ) {
+		if ( ! site.jetpack || site.versionCompare( '4.2-alpha', '<' ) ) {
 			return null;
 		}
 
 		return (
 			<JetpackSyncPanel />
 		);
-	},
+	}
 
 	render() {
-		const { site } = this.props;
+		const { isRequestingSettings, isSavingSettings, site, translate } = this.props;
 		if ( site.jetpack && ! site.hasMinimumJetpackVersion ) {
 			return this.jetpackDisconnectOption();
 		}
 
 		return (
-			<div className={ this.state.fetchingSettings ? 'is-loading' : '' }>
-				<SectionHeader label={ this.translate( 'Site Profile' ) }>
+			<div className={ isRequestingSettings ? 'is-loading' : '' }>
+				{ site && <QuerySiteSettings siteId={ site.ID } /> }
```

Comment:
>Another option would be to add a flag to the state when the request is performed at least once. But this was simpler. What do you think? shoud we add this flag?

I don't know that we need a state flag really. But one problem I see with this is what happens if the user starts changing fields and settings are then received by the mounting of `<QuerySiteSettings />`? It's non-ideal both if it interferes with the user's input, and if it doesn't that we're sending an unnecessary request and that the fields are potentially stale.

Now I'm curious though... what was the previous behavior?